catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -215,8 +222,88 @@ private org.apache.hadoop.hive.metastore.api.Partition toHivePartition(
   }
 
   @Override
-  public boolean dropPartition(String partitionName) {
-    throw new UnsupportedOperationException();
+  public boolean dropPartition(String partitionName, boolean ifExists)
+      throws NoSuchPartitionException {
+    try {
+      // check the partition exists
+      try {
+        table
+            .clientPool()
+            .run(c -> c.getPartition(table.schemaName(), table.name(), partitionName));
+      } catch (NoSuchObjectException e) {
+        if (ifExists) {
+          return true;
+        } else {
+          throw new NoSuchPartitionException(
+              e, "Hive partition %s does not exist in Hive Metastore", partitionName);
+        }
+      }
+
+      // get all partitions that will be deleted
+      Table hiveTable = table.clientPool().run(c -> c.getTable(table.schemaName(), table.name()));
+      List<org.apache.hadoop.hive.metastore.api.Partition> partitions =
+          table
+              .clientPool()
+              .run(
+                  c ->
+                      c.listPartitions(
+                          table.schemaName(),
+                          table.name(),
+                          getFilterPartitionList(hiveTable, partitionName),
+                          (short) -1));
+
+      // delete partitions iteratively
+      for (org.apache.hadoop.hive.metastore.api.Partition partition : partitions) {
+        table
+            .clientPool()
+            .run(
+                c ->
+                    c.dropPartition(
```

Comment:
You should add some comments here to clarify that the partition length is currently 1.

Reply:
this operation is for drop partition cascade 
for example :  there are partitions (log_date=2024-02-26/log_hour=01, log_date=2024-02-26/log_hour=02)
if user deliver log_date=2024-02-26, both of the partitions will be dropped


What happens if some drop operation fails and some succeed?
If one partition dropped failed, the partitions that have not been traversed after it will not be dropped, and will throw exception at here(partitions before it will be dropped), so the final result of this drop-operation will return failed


Why not support transactions?
I referred to Spark and HCat implementation, both of them did not support transactions when drop partition cascade.  I think the reason for this is to avoid increasing code complexity

spark:
<img width="835" alt="image" src="https://github.com/datastrato/gravitino/assets/58762426/d3b55ed7-6bdc-494e-87a4-d317f1baa651">

HCat:
<img width="901" alt="image" src="https://github.com/datastrato/gravitino/assets/58762426/acb07296-51c4-4451-a66d-c6fd268ebb93">